### ISSUE TITLE
Empty Payload

### ISSUE THREAD

**Issue by despia** [Influenzanet/authentication-service#1]
Using an empty payload causes the application to crash after trying to bind the payload using the gin frameworks

```go
context.ShouldBindJSON(&creds)
```

We need to further investigate what causes this and how to fix it.

As an intermediate fix for this crash I have commented out the corresponding test code and instead use

```go
t.Errorf("fix empty payload error")
return
```

to fail the test without crashing the application.

**Comment by phev8** [comment]
@despia what is the reason, you commented out the test code? As far as I know it doesn't influence other tests (at least on my machine, other tests pass well).

Generally, if a test causes the application to crash, it's critical to include this test in any future testing.

**Comment by despia** [comment]
The change has been reverted in 2dc9178.

**Comment by phev8** [comment]
To avoid this error, content length is checked at the very beginning of the method. Fixed in 79bdd57.
